vowpalwabbit/core/src/reductions/metrics.cc

```diff
@@ -137,38 +138,44 @@ void predict_or_learn(metrics_data& data, T& base, E& ec)
 
 void VW::reductions::output_metrics(VW::workspace& all)
 {
-  if (all.options->was_supplied("extra_metrics"))
+  metrics_manager& manager = all.global_metrics;
+  if (manager.are_metrics_enabled())
   {
-    std::string filename = all.options->get_typed_option<std::string>("extra_metrics").value();
-    VW::metric_sink list_metrics;
-
-    all.l->persist_metrics(list_metrics);
-
-    for (auto& metric_hook : all.metric_output_hooks) { metric_hook(list_metrics); }
+    all.l->persist_metrics(manager);
```

Comment:
This no longer actually collects metrics but registers callbacks. I don't think you should change the semantics of this call.

Reply:
reverted `persiste_metrics` and moved call to `collect_metrics`